File: source_code/java/src/br/com/codility/lessons/java/PermMissingElem.java
package br.com.codility.lessons.java;

public class PermMissingElem {

    public static int solution(int[] A) {

        int sum = 0;
        for (int i = 0; i < A.length; i++){
            sum += A[i];
            System.out.println("SUM: "+sum);
        }


        if (A.length % 2 == 0) {
            return -sum + (A.length / 2 + 1) *(A.length + 1);
        } else {
            return -sum + (A.length / 2 + 1) * (A.length + 2);
        }

        //return A.length % 2 == 0 ? -sum + (A.length/2 + 1) * (A.length+1)
        //       : -sum + (A.length/2 + 1) * (A.length+2);

    }

}

File: source_code/java/src/br/com/codility/lessons/kotlin/PermMissingElem.kt
package br.com.codility.lessons.kotlin

class PermMissingElem {

    companion object {
        fun solution(A: IntArray): Int {

            var sum = 0

            A.forEach {
                println("SUM: $sum")
                sum +=it
            }

            return if (A.size % 2 == 0) {
                -sum + (A.size / 2 + 1) * (A.size + 1)
            } else {
                -sum + (A.size / 2 + 1) * (A.size + 2)
            }

        }
    }

}